src/test/java/com/puppycrawl/tools/checkstyle/utils/CommonUtilTest.java

```diff
@@ -494,6 +494,56 @@ public void testGetUriByFilenameFindsResourceRelativeToRootClasspath() throws Ex
                 content, startsWith("good"));
     }
 
+    @Test
+    public void testGetCountOfEmptyLinesBeforeWithIllegalArguments() {
+        try {
+            CommonUtil.getCountOfEmptyLinesBefore(0, (String[]) null);
+            fail("IllegalArgumentException is expected");
+        }
+        catch (IllegalArgumentException ex) {
+            assertEquals("Invalid exception message", "Require non empty lines but got: null",
+                    ex.getMessage());
+        }
+
+        try {
+            CommonUtil.getCountOfEmptyLinesBefore(0);
+            fail("IllegalArgumentException is expected");
+        }
+        catch (IllegalArgumentException ex) {
+            assertEquals("Invalid exception message", "Require non empty lines but got: []",
+                    ex.getMessage());
+        }
+
+        try {
+            CommonUtil.getCountOfEmptyLinesBefore(0, "foo", "bar");
+            fail("IllegalArgumentException is expected");
+        }
+        catch (IllegalArgumentException ex) {
+            assertEquals("Invalid exception message",
+                    "Invalid lineNo: 0, expected lineNo range: [1, 2]", ex.getMessage());
+        }
+
+        try {
+            CommonUtil.getCountOfEmptyLinesBefore(3, "foo", "bar");
+            fail("IllegalArgumentException is expected");
+        }
+        catch (IllegalArgumentException ex) {
+            assertEquals("Invalid exception message",
+                    "Invalid lineNo: 3, expected lineNo range: [1, 2]", ex.getMessage());
+        }
+    }
+
+    @Test
+    public void testGetCountOfEmptyLinesBefore() {
+        final String[] lines = {"foo", "", "", "bar", "", "another"};
+        assertEquals("Invalid result", 0, CommonUtil.getCountOfEmptyLinesBefore(1, lines));
```

Comment:
@gmshake and @pbludov , I do not see explanation on this, that I requested at https://github.com/checkstyle/checkstyle/pull/6830#issuecomment-504688884 . Please share it.
Exceptional cases are ok, but testGetCountOfEmptyLinesBefore is not ok.

Reply:
For IT, see comment below :)

Reply:
No changes in CommonUtilTest.java is expected. Please move whole testing efforts to UT of Check, create as much Input files in resources as you need.